flow/api/auth/delete.go

```diff
@@ -0,0 +1,68 @@
+package auth
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	"flow/api/serde"
+	"flow/common/db"
+)
+
+const deleteAccountQuery = `
+DELETE FROM "user"
+WHERE id = $1
+`
+
+const deleteReviewsQuery = `
+DELETE FROM review
+WHERE user_id = $1
+`
+
+const selectAccountQuery = `
+SELECT email, first_name, last_name FROM "user" WHERE id = $1
+`
+
+func DeleteAccount(conn *db.Conn, w http.ResponseWriter, r *http.Request) error {
+	// Ensure user is authenticated
+	userId, err := serde.UserIdFromRequest(r)
+	if err != nil {
+		return serde.WithStatus(http.StatusUnauthorized, fmt.Errorf("extracting user id: %w", err))
+	}
+
+	// Parse query string to determine if reviews should be deleted
+	deleteReviews := r.URL.Query().Get("deleteReviews")
+	shouldDeleteReviews, err := strconv.ParseBool(deleteReviews)
+	if err != nil {
+		return serde.WithStatus(http.StatusBadRequest, fmt.Errorf("invalid deleteReviews parameter: %w", err))
+	}
+
+	var info userInfo
+	err = conn.QueryRow(selectAccountQuery, userId).Scan(&info.Email, &info.FirstName, &info.LastName)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return serde.WithStatus(http.StatusNotFound, fmt.Errorf("user id not found: %d", userId))
+		}
+		return fmt.Errorf("fetching user data: %w", err)
+	}
+
+	// Optionally delete user reviews
+	if shouldDeleteReviews {
+		tag, err := conn.Exec(deleteReviewsQuery, userId)
+		if err != nil {
+			return fmt.Errorf("deleting user reviews: %w", err)
+		}
+		log.Printf("Deleted %d reviews for user %d", tag.RowsAffected(), userId)
+	}
```

Comment:
Not sure if we need this option tbh but if we want to keep it, I'd wrap this + the delete account query into a single DB transaction so we don't end up in a state where the account exists but all their reviews got deleted in case of error

Reply:
I think you can just make your life easier and not delete reviews for now, and then merge it in, no one's ever requested to delete their reviews before, they only care about their personal course data etc.